STM32F1/libraries/A_STM32_Examples/examples/Sensors/HardTimerAsEncoder/HardTimerAsEncoder.ino

```diff
@@ -26,92 +26,53 @@
  * mode 2 - only counts pulses on Channel A
  * mode 3 - counts on both channels. 
  * 
-*/
-
-
+ */
 
 #include "HardwareTimer.h"
 
 //Encoder simulation stuff
 //NOT NEEDED WHEN CONNECTING A REAL ENCODER
-unsigned char mode = 0;  //to issue steps...
-unsigned char dir = 'F'; // direction of movement of the encoder.
-
-unsigned int freq = 100; //update frequency.
-unsigned long time = 0;        //time variable for millis()
-unsigned char states[4];       //because I'm lazy...
 
+const uint8_t output_a = PA7; // D4
+const uint8_t output_b = PA6; // D5
 
 //Encoder stuff
+const uint8_t input_a = PB6; // D16 this should be the pin of timer 4 channel 1
+const uint8_t input_b = PB7; // D15 this should be the pin of timer 4 channel 2
 
 //Pulses per revolution
 #define PPR   1024
 
-HardwareTimer timer(2);
+HardwareTimer timer(4);
 
-unsigned long ints = 0;
+volatile unsigned long revolutions = 0;
 
```

Comment:
I know that the original uses unsigned long, but I think an `uint32_t` would also satisfy the requirements.

Reply:
yes, In embedded systems I prefer to be specific with the data types. I'll change to uint32_t, even though they are technically the same.